tests/base/symmetric_tensor_39.cc

```diff
@@ -0,0 +1,69 @@
+#include <iostream>
+#include <fstream>
+
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/vectorization.h>
+
+using namespace dealii;
+
+int main()
+{
+  std::string logname = "output";
+  std::ofstream logfile(logname);
+
+#if defined DEAL_II_COMPILER_VECTORIZATION_LEVEL
+  std::cout << "Vectorization preprocessor variable found!" << std::endl;
+#endif
+
+#define INVALID_INT_REPRESENTATION 0
+#define SSE_INT_REPRESENTATION 1
+#define AVX_INT_REPRESENTATION 2
+
+#if DEAL_II_COMPILER_VECTORIZATION_LEVEL == INVALID_INT_REPRESENTATION
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents no vectorization" << std::endl;
+#elif DEAL_II_COMPILER_VECTORIZATION_LEVEL == SSE_INT_REPRESENTATION
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents SSE" << std::endl;
+#elif DEAL_II_COMPILER_VECTORIZATION_LEVEL == AVX_INT_REPRESENTATION
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL represents AVX" << std::endl;
+#else
+  std::cout << "Value of DEAL_II_COMPILER_VECTORIZATION_LEVEL not in {0,1,2}" << std::endl;
+#endif
+
+  const unsigned int dim = 3;
+
+  SymmetricTensor<4,dim,VectorizedArray<double>> I;
+  SymmetricTensor<2,dim,VectorizedArray<double>> A;
+  SymmetricTensor<2,dim,VectorizedArray<double>> B;
+
+  // While at it testing that different constructors / assignments work.
+  VectorizedArray<double> zero(0.0);
+  VectorizedArray<double> one = make_vectorized_array(1.0);
+  VectorizedArray<double> half;
+  half = 0.5;
+
+  // I^sym = 0.5(d_ik*d_jl + d_il*d_jk) -> I^sym : A = A^sym
+  for (unsigned int i = 0; i < dim; i++)
+    for (unsigned int j = 0; j < dim; j++)
+      for (unsigned int k = 0; k < dim; k++)
+        for (unsigned int l = 0; l < dim; l++)
+          I[i][j][k][l] = ( (i == k && j== l && i == l && j == k) ? one : ( (i == k && j== l) || (i == l && j == k) ? half : zero));
+
+  for (unsigned int i = 0; i < dim; ++i)
+    for (unsigned int j = 0; j < dim; ++j)
+      A[i][j] = make_vectorized_array(42.0);
```

Comment:
maybe make it different for different `i` and `j` and each `VectorizedArray<double>::n_array_elements`? Just to make sure it does what you expect.